From d3f2e15ea229dc9cbb61a6d4edb49d7b7bc4c144 Mon Sep 17 00:00:00 2001
From: Nathan Chancellor <natechancellor@gmail.com>
Date: Mon, 8 Oct 2018 15:42:15 -0700
Subject: [PATCH 13/15] DO-NOT-UPSTREAM: efi/libstub: Add missing compiler
 flags to hide warnings on x86_64

-Wpointer-sign has been disabled since the beginning of git history.

-Waddress-of-packed-member was disabled in commit bfb38988c51e ("kbuild:
clang: Disable 'address-of-packed-member' warning").

-Wgnu* was disabled in commit 61163efae020 ("kbuild: LLVMLinux: Add
Kbuild support for building kernel with Clang").

Signed-off-by: Nathan Chancellor <natechancellor@gmail.com>
---
 drivers/firmware/efi/libstub/Makefile | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/firmware/efi/libstub/Makefile b/drivers/firmware/efi/libstub/Makefile
index 7eb1bf753bbf..e433d5ee633a 100644
--- a/drivers/firmware/efi/libstub/Makefile
+++ b/drivers/firmware/efi/libstub/Makefile
@@ -9,7 +9,10 @@ cflags-$(CONFIG_X86_32)		:= -march=i386
 cflags-$(CONFIG_X86_64)		:= -mcmodel=small
 cflags-$(CONFIG_X86)		+= -m$(BITS) -D__KERNEL__ -O2 \
 				   -fPIC -fno-strict-aliasing -mno-red-zone \
-				   -mno-mmx -mno-sse -fshort-wchar
+				   -mno-mmx -mno-sse -fshort-wchar \
+				   -Wno-pointer-sign \
+				   $(call cc-disable-warning, address-of-packed-member) \
+				   $(call cc-disable-warning, gnu)
 
 # arm64 uses the full KBUILD_CFLAGS so it's necessary to explicitly
 # disable the stackleak plugin
-- 
2.19.1

